static/client/common.js

```diff
@@ -287,3 +287,15 @@ CCC.Common.createSvgElement = function(name, attrs, opt_parent) {
   }
   return el;
 };
+
+/**
+ * Given plain text, encode spaces and tabs such that HTML won't crush it.
+ * Does not handle line breaks in any way.
+ * @param {string} text Plain text.
+ * @return {string} HTML with any runs of spaces encoded.
+ */
+CCC.Common.escapeSpaces = function(text) {
+  return text.replace(/\t/g, '\u00A0 \u00A0 ')
```

Comment:
Tabs are conventionally eight spaces, not four.  (That's leaving aside the issue of actual tabular alignment.)

Also: wouldn't `&nbsp;` be clearer?

Reply:
Changed to 8.

&nbsp; is HTML (suitable for .innerHTML assignment), we need the equivalent Unicode character (suitable for document.createTextNode useage).